node_helper.js

```diff
@@ -9,61 +9,66 @@ const Spotify = require("./Spotify.js")
+    this.config = null
+    this.spotifyConfig = null
+    this.spotify = null
+    this.timer = null
+    this.firstStart = true
   },
 
-  initAfterLoading: function (config) {
-    this.config = config
-    let file = path.resolve(__dirname, "spotify.config.json");
-    if (fs.existsSync(file)) {
-      let parsedConfigurations = JSON.parse(fs.readFileSync(file));
-      this.spotifyConfigurations = updateOldSingleSpotifyConfigurationToNewMultipleSpotifyConfiguration(parsedConfigurations);
-      this.spotifyConfigurations.forEach(configuration => {
-        this.spotifies.push(new Spotify(configuration, this.config.debug));
-      });
+  doSpotifyConfig: function (configuration, account) {
+    if (!isNaN(account) && Array.isArray(configuration)) {
+      return configuration[account] // only wanted account or first
+    }
+    if (Array.isArray(configuration)) {
+      var found
+      configuration.forEach((jsAccount,number) => {
+        if (jsAccount.USERNAME == account) found = number
+      })
+      if (typeof found !== "undefined") return configuration[found]
+      else return configuration[0]
     }
-    this.findCurrentSpotify().then(() => {
-      console.log("[SPOTIFY] Started")
-      if (this.config.onStart) this.onStart()
-    });
+    // not update required not an array (single account)
+    return configuration
   },
 
-  findCurrentSpotify: async function () {
-    let playing = false;
-    for (let spotify of this.spotifies) {
-      this.spotify = spotify;
-      playing = true;
+  initAfterLoading: function (config, account) {
+    this.config = config
+    if (!account) {
+      account = this.config.accountDefault
+      console.log("[SPOTIFY] MMM-Spotify Version:",  require('./package.json').version)
+    }
+    let file = path.resolve(__dirname, "spotify.config.json")
+    if (fs.existsSync(file)) {
       try {
-        let result = await this.updateSpotify(spotify);
-        this.sendSocketNotification("CURRENT_PLAYBACK", result);
+        this.spotifyConfig = this.doSpotifyConfig(JSON.parse(fs.readFileSync(file)), account)
       } catch (e) {
-        //console.log("[SPOTIFY] This spotify is not playing:", spotify.config.USERNAME)
+        return console.log("[SPOTIFY] ERROR: spotify.config.json", e.name)
       }
+      if (!this.spotifyConfig) return console.log("[SPOTIFY] ERROR: Account not found")
+      this.spotify = new Spotify(this.spotifyConfig, this.config.debug)
     }
-    if (!playing) {
-      this.sendSocketNotification("CURRENT_PLAYBACK_FAIL");
-      setTimeout(() => {
-        this.findCurrentSpotify();
-      }, this.config.updateInterval);
-    } else {
-      setTimeout(() => {
-        this.updatePulse()
-      }, this.config.updateInterval)
+    else return console.log("[SPOTIFY] ERROR: spotify.config.json file missing !")
+    this.updatePulse().then(() => {
+      console.log("[SPOTIFY] Started with Account:", (this.spotify.config.USERNAME ? this.spotify.config.USERNAME : "default"))
```

Comment:
do we need this log?

Reply:
for error no, i whould be displayed

maybe:

if (this.config.debug) console.log("[SPOTIFY] Started with Account:", (this.spotify.config.USERNAME ? this.spotify.config.USERNAME : "default"))